src/metabase/models/dashboard_card.clj

```diff
@@ -109,23 +109,23 @@
   [{:keys [id series parameter_mappings visualization_settings] :as dashboard-card} :- DashboardCardUpdates]
   (let [{:keys [sizeX sizeY row col series]} (merge {:series []} dashboard-card)]
     (db/transaction
-      ;; update the dashcard itself (positional attributes)
-      (when (and sizeX sizeY row col)
-        (db/update-non-nil-keys! DashboardCard id
-          :sizeX                  sizeX
-          :sizeY                  sizeY
-          :row                    row
-          :col                    col
-          :parameter_mappings     parameter_mappings
-          :visualization_settings visualization_settings))
-      ;; update series (only if they changed)
-      (when-not (= series (map :card_id (db/select [DashboardCardSeries :card_id]
-                                          :dashboardcard_id id
-                                          {:order-by [[:position :asc]]})))
-        (update-dashboard-card-series! dashboard-card series))
-      ;; fetch the fully updated dashboard card then return it (and fire off an event)
-      (->> (retrieve-dashboard-card id)
-           (events/publish-event! :dashboard-card-update)))))
+     ;; update the dashcard itself (positional attributes)
+     (when (and sizeX sizeY row col)
+       (db/update-non-nil-keys! DashboardCard id
+                                :sizeX                  sizeX
+                                :sizeY                  sizeY
+                                :row                    row
+                                :col                    col
+                                :parameter_mappings     parameter_mappings
+                                :visualization_settings visualization_settings))
+     ;; update series (only if they changed)
+     (when-not (= series (map :card_id (db/select [DashboardCardSeries :card_id]
+                                                  :dashboardcard_id id
+                                                  {:order-by [[:position :asc]]})))
+       (update-dashboard-card-series! dashboard-card series)))
+    (u/prog1 (retrieve-dashboard-card id)
+      ;; todo: should this have an actor_id from the api user?
+      (events/publish-event! :dashboard-card-update <>))))
```

Comment:
Do you still need the `prog1` here?

Reply:
yes if `<>` needs an actor_id assoc'd to it and not otherwise. Good catch. I'm _pretty_ sure `:actor_id` needs to be on there but it wasn't in the past.

Reply:
So interestingly, there is no event handler for `:dashboard-card-update` or `:dashboard-card-create`. And there hasn't been for the five years of its existence. I'm not sure how dashboard-cards and dashboards interact for the purposes of revisions. I wonder if we can properly revert them in all cases. Either way those events can be removed for now.

Reply:
We cannot revert properly for all revisions - sometimes fails, other time creates some weird corruption - that's issue 6884